src/settings.rs

```diff
@@ -0,0 +1,37 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+// Describe the settings your policy expects when
+// loaded by the policy server.
+#[derive(Serialize, Deserialize, Default, Debug)]
+#[serde(default)]
+pub(crate) struct Settings {
+    #[serde(default)]
+    pub allowed_profiles: HashSet<String>,
+    pub profile_types: HashSet<String>,
+    pub localhost_profiles: HashSet<String>,
+}
+
+impl kubewarden::settings::Validatable for Settings {
+    fn validate(&self) -> Result<(), String> {
+        for typee in &self.profile_types {
+            if !vec![
+                "Localhost".to_string(),
+                "RuntimeDefault".to_string(),
+                "Unconfined".to_string(),
+            ]
```

Comment:
This should be declared outside of the `for` loop. Also I would use a Set instead of that vector, it has better lookup performances.

It's a bit ridiculous in this context to look for extra optimizations... so take that as a nitpicking comment

Reply:
I agree. You are right.